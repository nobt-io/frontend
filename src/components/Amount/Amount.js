import React from "react";
import { FormattedNumber } from "react-intl";
import { connect } from "react-redux";
import { getCurrency } from "../../routes/App/modules/currentNobt/selectors";
import classnames from "classnames";

const valueViewModel = (props) => (props.absolute) ? Math.abs(props.value): props.value;

const computeClassNames = (props) => classnames({
  [`${props.theme.root}`]: props.theme.root != null,
  [`${props.theme.positive}`]: props.theme.positive != null && props.value > 0,
  [`${props.theme.negative}`]: props.theme.negative != null && props.value < 0,
});

const Amount = (props) => (
  <span className={computeClassNames(props)}>
    <FormattedNumber value={valueViewModel(props)} currency={props.currency} style="currency"/>
  </span>
);

Amount.defaultProps = {
  absolute: true,
  theme: {
    root: null,
    positive: null,
    negative: null
  }
};

Amount.propTypes = {
  value: React.PropTypes.number.isRequired,
  absolute: React.PropTypes.bool,
  theme: React.PropTypes.shape({
    root: React.PropTypes.string,
    positive: React.PropTypes.string,
    negative: React.PropTypes.string
  })
};

export default connect(
  (state) => ({
    currency: getCurrency(state)
  }),
  () => ({})
)(Amount)
