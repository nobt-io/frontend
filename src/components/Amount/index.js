import Amount from './Amount';

export default Amount;
