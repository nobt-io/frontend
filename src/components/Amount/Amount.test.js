import React from "react";
import renderer from "react-test-renderer";
import Amount from "components/Amount/index";
import { Provider } from "react-redux";
import createMockStore from "../../store/mockStore";
import { IntlProvider } from "react-intl";

test('should use absolute value of amount by default', () => {

  let mockStore = createMockStore();

  let cmp = renderer.create(
    <IntlProvider>
      <Provider store={mockStore}>
        <Amount value={-30} />
      </Provider>
    </IntlProvider>
  );

  let tree = cmp.toJSON();

  expect(tree).toMatchSnapshot();
});

test('should display actual value if absolute property is falsy', () => {

  let mockStore = createMockStore();

  let cmp = renderer.create(

    <IntlProvider>
      <Provider store={mockStore}>
        <Amount value={-30} absolute={false}/>
      </Provider>
    </IntlProvider>
  );

  let tree = cmp.toJSON();

  expect(tree).toMatchSnapshot();
});
