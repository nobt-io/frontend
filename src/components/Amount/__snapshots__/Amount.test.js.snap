// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display actual value if absolute property is falsy 1`] = `
<span
  className=""
>
  <span>
    -€30.00
  </span>
</span>
`;

exports[`should use absolute value of amount by default 1`] = `
<span
  className=""
>
  <span>
    €30.00
  </span>
</span>
`;
